Tests

xcixor opened the pull request:
#### Purpose of this PR
The purpose of this PR is to write failing tests that define the desired features of the application

#### Tasks to be completed
* Write tests for api endpoints 
* Write tests for Ride model
* Write tests for User mode
* Write tests for Controller 

#### Manual Testing
* Clone this repo
* Check out to the tests branch 
* Create a virtual env and install (pytests)[https://docs.pytest.org/en/3.2.5/getting-started.html]
* Run pytest tests/ to run all the tests in the tests directory

#### Relevant Pivotal Tracker stories
- [Write tests for application #158575425](https://www.pivotaltracker.com/n/projects/2179559)

xcixor committed:
chore(folder structure)

-Create folder structure for the api application

xcixor committed:
Merge branch 'develop' into tests

xcixor committed:
tests(api endpoints)

-Add minimum endpoints for api

xcixor committed:
[#158575425] tests(api endpoints)

-Add more tests for api endpoints

xcixor committed:
[#158575425] tests(api endpoints)

-Add tests for Ride class.

xcixor committed:
[#158575425] tests(api endpoints)

-Add tests for User class.

xcixor committed:
[#158575425] tests(api endpoints)

-Add tests for the controller.

xcixor committed:
[Finishes #158575425]

JoshuaOndieki reviewed (changes requested):
Here's how you can better name your tests :arrow_right: http://osherove.com/blog/2005/4/3/naming-standards-for-unit-tests.html

TL;DR
- Test name should express a specific requirement.
- Test name should include the expected input or state and the expected result for that input or state.
- Test name should be presented as a statement or fact of life that expresses workflows and outputs.
- Test Name should only begin with Test if it is required by the testing framework or if it eases development and maintenance of the unit tests in some way.
- Test name should include the name of tested method or class.
- Variable names should express the expected input and state.

xcixor commented:
Thanks for that insight, Let me review that @JoshuaOndieki

xcixor committed:
refactor(test function names)

Rename tests to describe to follow the convention, MethodName_StateUnderTest_ExpectedBehavior, to ease readability and explain expected feature behavior.

xcixor committed:
refactor(api endpoints)

-Add authorization token for tests.

JoshuaOndieki reviewed (approved):
:+1: Great effort on the test naming.

xcixor committed:
ignore pytest cache

xcixor committed:
refactor(tests folder structure)

-Restructure folder structure so that tests for a certain unit are contained in the same folder
